src/gt4py/next/ffront/fbuiltins.py

```diff
@@ -95,6 +95,9 @@ def __call__(self, *args: _P.args, **kwargs: _P.kwargs) -> _R:
 
     def dispatch(self, *args: Any) -> Callable[_P, _R]:
         arg_types = tuple(type(arg) for arg in args)
+        if self.function == where.function and arg_types[1] == tuple and arg_types[2] == tuple:
+            # TODO(havogt): temporary workaround for `where(cond, tuple(...), tuple(...))`
```

Comment:
In the end I didn't like this fall-back proposal, I would like to keep this special case and then revisit the pattern once we have all builtins implemented.